airsonic-main/src/main/java/org/airsonic/player/domain/PlayQueue.java

```diff
@@ -311,17 +311,7 @@ public synchronized void rearrange(int[] indexes) {
      * @param index The playlist index.
      */
     public synchronized void moveUp(int index) {
-        makeBackup();
-        if (index <= 0 || index >= size()) {
-            return;
-        }
-        Collections.swap(files, index, index - 1);
-
-        if (this.index == index) {
-            this.index--;
-        } else if (this.index == index - 1) {
-            this.index++;
-        }
+        moveDown(index - 1);
```

Comment:
Reentrant locks in Java, so this should be okay without any deadlock possibility.

logic looks okay to me